### PR TITLE
Add checking available resources

### PR DESCRIPTION
**Pull request by andkazakov**
**Problem description**
The "deprecated checker" now checks versions and resource availability based on a [static file](https://github.com/2gis/k8s-handle/blob/master/k8s_handle/k8s/deprecation_checker.py). But in kubernetes settings we can set the available API groups via config file. See [example](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.16.md#deprecations-and-removals). 

**Proposed solution**
1. check resource availability bases on API responses k8s
2. returns error if resource is not found in API
3. deprication checker should show warnings in logs only

### CODE REVIEW

File: k8s_handle/k8s/availability_checker/checker.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+                kinds = api.get_resources_by_version(api_group)
+
+        return kind in kinds
+
+    def run(self, file_path: str):
+        for template_body in get_template_contexts(file_path):
+            available = self._is_available_kind(template_body.get('apiVersion'), template_body.get('kind'))
+
+            if not available:
+                msg = "The resource with kind {} is not supported with version {}. File: {}"
```
Comment (furiousassault) [inline k8s_handle/k8s/availability_checker/checker.py]: And `msg` too. "foo {} bar {}".format(args...)?

File: k8s_handle/k8s/availability_checker/resource_getters.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+
+class CoreResourceGetter(AbstractResourceGetter):
+
+    def __init__(self, resource_api: CoreResourcesAPI):
+        self.api = resource_api
+        self.versions = {}
+
+    def is_processable_version(self, api_group: str) -> bool:
+        return len(api_group.split(VERSION_DELIMITER, 1)) == 1
```
Comment (furiousassault) [inline k8s_handle/k8s/availability_checker/resource_getters.py]: https://docs.python.org/3/library/stdtypes.html#str.count
This may simplify the check.

Comment (furiousassault) [reply k8s_handle/k8s/availability_checker/resource_getters.py]: Or, maybe we can do split and check that split parts is not empty, to not to go to API if something like "/" or empty string is given. And add that cases in unit test.

File: k8s_handle/k8s/api_extensions.py [hunk 3]
```diff
@@ -42,3 +42,40 @@ def list_api_resource_arbitrary(self, group, version):
 
             log.error('{}'.format(add_indent(e.body)))
             raise ProvisioningError(e)
+
+
+class CoreResourcesAPI(client.CoreApi):
+
+    def list_api_resources(self, version):
+        try:
+            return self.api_client.call_api(
```
Comment (rvadim) [inline k8s_handle/k8s/api_extensions.py]: Why we set all parameters for this api call? What reason? Most of parameters have default values.

File: k8s_handle/__init__.py [hunk 4]
```diff
@@ -106,9 +107,11 @@ def _handler_provision(command, resources, priority_evaluator, use_kubeconfig, s
 
     try:
         deprecation_checker = ApiDeprecationChecker(client.VersionApi().get_code().git_version[1:])
+        available_checker = ResourceAvailabilityChecker(make_resource_getters_list())
 
         for resource in resources:
             deprecation_checker.run(resource)
+            available_checker.run(resource)
```
Comment (rvadim) [inline k8s_handle/__init__.py]: Any options to disable it? What if user don't have permissions to run this kind of API path?

**Comment by rvadim** [comment]
What do you think about adding some documentation about this feature?

File: k8s_handle/__init__.py [hunk 5]
```diff
@@ -228,11 +231,13 @@ def main():
...
         sys.exit(1)
     except DeprecationError as e:
-        log.error('Deprecation warning: {}'.format(e))
-        sys.exit(1)
+        log.warning('Deprecation warning: {}'.format(e))
     except RuntimeError as e:
         log.error('RuntimeError: {}'.format(e))
         sys.exit(1)
+    except ResourceNotAvailableError as e:
+        log.error('{}'.format(e))
```
[the remark this answers is not in the document]
Comment (andkazakov) [reply k8s_handle/__init__.py]: fixed

File: k8s_handle/__init__.py (hunk 4 above)
Comment (andkazakov) [reply k8s_handle/__init__.py]: Today I checked whether I can restrict access for this API (api/{version} or /apis/{group}/version). 
Created token with access to get only Pod, but was still able to get a list of available versions.

Can you tell me, how set limitation for this API method? @seleznev @furiousassault

Comment (rvadim) [reply k8s_handle/__init__.py]: Ok, `api` and `apis` available w/o auth as I see.

**Pull request merged**